src/main/java/org/broadinstitute/hellbender/utils/read/ReadQueryNameComparator.java

```diff
@@ -0,0 +1,61 @@
+package org.broadinstitute.hellbender.utils.read;
+
+import htsjdk.samtools.SAMRecordQueryNameComparator;
+import htsjdk.samtools.SAMTag;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+/**
+ * compare {@link GATKRead} by queryname
+ * duplicates the exact ordering of {@link SAMRecordQueryNameComparator}
```

Comment:
Duplication of code from HTSJDK to GATK (and other downstream projects) is a sign that htsjdk needs to evolve to an interface-based - as @lbergelson is a maintainer, I point out again that #4340 and htsjdk-related issues should be a high priority!

Reply:
@magicDGS You're very right.  We had hoped to begin work on a major htsjdk revamp last quarter, but we didn't have enough bandwidth to take it on along with other requirements we had.  We're hoping to start very soon.